pet.el

```diff
@@ -923,15 +911,74 @@ FN is `eglot--guess-contact', ARGS is the arguments to
                 (seq-subseq result 4))
       result)))
 
+(defun pet-eglot-alternatives (alternatives)
+  "Use `pet-executable-find' instead of `executable-find' in the
+`eglot-alternatives' closure."
+  (lambda (&optional interactive _project)
```

Comment:
By "replacing the definition of `executable-find`, I don't mean copy and paste, there's `cl-flet` and `cl-letf` for these kind of things.

Reply:
I've kept the structure similar to the existing eglot code on purpose, because it makes reading and understanding the code simpler in my opnion. 

Even if we were to use `cl-flet`, we should still have to reproduce the entire lambda starting on line 917 because that's what the closure is. Only one line in the closure changes (replacing of `executable-find` with `pet-executable-find`, on line 932), but we have to copy the entire closure over anyway because the rest of the logic is still necessary.